odoo/orm/fields_numeric.py

```diff
@@ -208,7 +208,7 @@ def convert_to_column_insert(self, value, record, values=None, validate=True):
             # currencies, which is functional nonsense and should not happen
             # BEWARE: do not prefetch other fields, because 'value' may be in
             # cache, and would be overridden by the value read from database!
-            currency = record[:1].with_context(prefetch_fields=False)[currency_field_name]
+            currency = record[:1].sudo().with_context(prefetch_fields=False)[currency_field_name]
```

Comment:
It makes sense. Could you add a test for it? Pretty sure that you can have the issue even without the cache poisoning.

Reply:
Test added